astropy/modeling/mappings.py

```diff
@@ -34,8 +38,8 @@ def mapping(self):
         return self._mapping
 
     def evaluate(self, *args):
-        if len(args) < self.n_inputs:
-            raise TypeError('{0} expects at most {1} inputs; got {2}'.format(
+        if len(args) != self.n_inputs:
+            raise TypeError('{0} expects {1} inputs; got {2}'.format(
```

Comment:
For swapping, just as an example, you didn't *have* to specify all indices.  For example, you could write something like `Mapping((1, 0))` and that would swap the first and second inputs, but pass any others (2 or above) through.  I don't have any strong feelings about whether that's preferable though.